packages/openapi-v3/src/parameter-decorator.ts

```diff
@@ -46,263 +27,498 @@ const paramDecoratorStyle = Symbol('ParamDecoratorStyle');
  *   ) {}
  * }
  * ```
- * Please note mixed usage of `@param` at method/parameter level is not allowed.
  *
  * @param paramSpec Parameter specification.
  */
 export function param(paramSpec: ParameterObject) {
   return function(
     target: Object,
     member: string | symbol,
-    descriptorOrIndex: TypedPropertyDescriptor<ExtensionValue> | number,
+    // Comment for review purpose:
+    // deprecate method level decorator
+    index: number,
   ) {
     paramSpec = paramSpec || {};
     // Get the design time method parameter metadata
     const methodSig = MetadataInspector.getDesignTypeForMethod(target, member);
```

Comment:
I think this is a remnant of having method level `@param` decorator. Can we rework the code here so that we don't need leverage `getDesignTypeForMethod` here?

Reply:
@shimks interestingly, I had the same thought, but I am afraid that's the only inspector method we can use, and it makes perfect sense.
If you look into all inspector methods in `metadata/src/inspector.ts`, this is the only function that returns design type information of a method, including its parameter types, return types(we can use it for generating response object in the future).

Reply:
ah, but is `ResponseObject` something we want to generate from `@param` or `@requestBody` in the future? I guess that's up for discussion way down the road, but I guess it's ok to leave it in for now since it's not a huge issue anyway.